akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -308,7 +309,30 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
   @volatile private[this] var aeronErrorLogTask: Cancellable = _
   @volatile private[this] var areonErrorLog: AeronErrorLog = _
 
+  // Compression 
+  /**
+   * Compression tables must be created once, such that inbound lane restarts don't cause dropping of the tables.
+   * However are the InboundCompressions are owned by the Decoder stage, and any call into them must be looped through the Decoder!
+   *
+   * Use `inboundCompressionAccess` (provided by the materialized `Decoder`) to call into the compression infrastructure.
+   */
+  private[this] val _inboundCompressions = new AtomicReference[InboundCompressions] // atomic since restarts come from Futures
```

Comment:
I think it would be enough with `volatile var`, but is it even possible to create it in constructor and just use a `val`?